File: static/java/GCD.java
import java.math.BigInteger;
import java.util.Random;

public class GCD {
  private static long gcdGolden(long a, long b) {
    BigInteger b1 = BigInteger.valueOf(a);
    BigInteger b2 = BigInteger.valueOf(b);
    BigInteger gcd = b1.gcd(b2);
    return gcd.longValue();
  }

  private static void check(long a, long b) {
    assert GCD1.GCD(a, b) == gcdGolden(a, b);
  }

  public static void main(String[] args) {
    check(2L, 2L);
    check(2L, 3L);
    check(17L, 289L);
    check(13L, 17L);
    check(17L, 289L);
    check(18L, 24L);
    check(1024L * 1023L, 1023L * 1025L);
    check(317L * 1024L * 1023L, 317L * 1023L * 1025L);
    check(Long.MAX_VALUE, Long.MAX_VALUE - 1L);
    check(Long.MAX_VALUE - 1L, (Long.MAX_VALUE - 1L) / (2L));
    check(0L, 0L);
    check(0L, 1L);
    check(10L, 100L);
    long x = 18, y = 12;
    assert(GCD1.GCD(x, y) == 6);
    if (args.length == 2) {
      x = Integer.parseInt(args[0]);
      y = Integer.parseInt(args[1]);
      System.out.println(GCD1.GCD(x, y));
      assert(GCD1.GCD(x, y) == GCD2.GCD(x, y));
    } else {
      Random r = new Random();
      for (int times = 0; times < 1000; ++times) {
        x = r.nextInt(Integer.MAX_VALUE);
        y = r.nextInt(Integer.MAX_VALUE);
        assert(GCD1.GCD(x, y) == GCD2.GCD(x, y));
      }
    }
  }
}

File: static/python/gcd.py
import sys
import random
import fractions
import gcd1
import gcd2


def check(a, b):
    assert gcd1.gcd(a, b) == gcd2.gcd(a, b)


def main():
    check(2, 2)
    check(2, 3)
    check(17, 289)
    check(13, 17)
    check(17, 289)
    check(18, 24)
    check(1024 * 1023, 1023 * 1025)
    check(317 * 1024 * 1023, 317 * 1023 * 1025)
    check(sys.maxsize, sys.maxsize - 1)
    check(sys.maxsize - 1, (sys.maxsize - 1) // 2)
    check(0, 0)
    check(0, 1)
    check(10, 100)
    x = 18
    y = 12
    assert gcd1.gcd(x, y) == 6
    if len(sys.argv) == 3:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
        print(gcd1.gcd(x, y))
        assert gcd1.gcd(x, y) == fractions.gcd(x, y) == gcd2.gcd(x, y)
    else:
        for _ in range(1000):
            x = random.randint(1, sys.maxsize)
            y = random.randint(1, sys.maxsize)
            assert gcd1.gcd(x, y) == fractions.gcd(x, y) == gcd2.gcd(x, y)


if __name__ == '__main__':
    main()
